diff --git a/node_modules/@dhis2/pwa/build/cjs/index.js b/node_modules/@dhis2/pwa/build/cjs/index.js
index e062c8c..2e88e4d 100644
--- a/node_modules/@dhis2/pwa/build/cjs/index.js
+++ b/node_modules/@dhis2/pwa/build/cjs/index.js
@@ -9,6 +9,12 @@ Object.defineProperty(exports, "OfflineInterface", {
     return _offlineInterface.OfflineInterface;
   }
 });
+Object.defineProperty(exports, "OfflineInterfaceProvider", {
+  enumerable: true,
+  get: function () {
+    return _OfflineInterfaceContext.OfflineInterfaceProvider;
+  }
+});
 Object.defineProperty(exports, "REGISTRATION_STATE_ACTIVE", {
   enumerable: true,
   get: function () {
@@ -69,6 +75,18 @@ Object.defineProperty(exports, "setUpServiceWorker", {
     return _setUpServiceWorker.setUpServiceWorker;
   }
 });
+Object.defineProperty(exports, "useOfflineInterface", {
+  enumerable: true,
+  get: function () {
+    return _OfflineInterfaceContext.useOfflineInterface;
+  }
+});
+Object.defineProperty(exports, "usePWAUpdateState", {
+  enumerable: true,
+  get: function () {
+    return _usePWAUpdateState.usePWAUpdateState;
+  }
+});
 
 var _setUpServiceWorker = require("./service-worker/set-up-service-worker.js");
 
@@ -77,3 +95,7 @@ var _offlineInterface = require("./offline-interface/offline-interface.js");
 var _registration = require("./lib/registration.js");
 
 var _baseUrlDb = require("./lib/base-url-db.js");
+
+var _OfflineInterfaceContext = require("./react/OfflineInterfaceContext.js");
+
+var _usePWAUpdateState = require("./react/usePWAUpdateState.js");
\ No newline at end of file
diff --git a/node_modules/@dhis2/pwa/build/cjs/react/OfflineInterfaceContext.js b/node_modules/@dhis2/pwa/build/cjs/react/OfflineInterfaceContext.js
new file mode 100644
index 0000000..27e3b7a
--- /dev/null
+++ b/node_modules/@dhis2/pwa/build/cjs/react/OfflineInterfaceContext.js
@@ -0,0 +1,49 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.useOfflineInterface = exports.OfflineInterfaceProvider = void 0;
+
+var _propTypes = _interopRequireDefault(require("prop-types"));
+
+var _react = _interopRequireWildcard(require("react"));
+
+var _offlineInterface = require("../offline-interface/offline-interface.js");
+
+function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
+
+function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
+
+function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
+
+const OfflineInterfaceContext = /*#__PURE__*/(0, _react.createContext)(null); // nb: instantiating an offline interface here creates a side-effect where
+// it is bundled in the service worker, which causes an error when the `window`
+// API is referenced.
+// It seems like some react and prop types leaks into the SW as well, which is
+// interesting because other exports like from registration.js don't
+
+let theOfflineInterface = null;
+
+const OfflineInterfaceProvider = _ref => {
+  let {
+    children
+  } = _ref;
+
+  if (theOfflineInterface === null) {
+    theOfflineInterface = new _offlineInterface.OfflineInterface();
+  }
+
+  return /*#__PURE__*/_react.default.createElement(OfflineInterfaceContext.Provider, {
+    value: theOfflineInterface
+  }, children);
+};
+
+exports.OfflineInterfaceProvider = OfflineInterfaceProvider;
+OfflineInterfaceProvider.propTypes = {
+  children: _propTypes.default.node
+};
+
+const useOfflineInterface = () => (0, _react.useContext)(OfflineInterfaceContext);
+
+exports.useOfflineInterface = useOfflineInterface;
\ No newline at end of file
diff --git a/node_modules/@dhis2/pwa/build/cjs/react/usePWAUpdateState.js b/node_modules/@dhis2/pwa/build/cjs/react/usePWAUpdateState.js
new file mode 100644
index 0000000..8e64773
--- /dev/null
+++ b/node_modules/@dhis2/pwa/build/cjs/react/usePWAUpdateState.js
@@ -0,0 +1,64 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.usePWAUpdateState = void 0;
+
+var _react = require("react");
+
+var _OfflineInterfaceContext = require("./OfflineInterfaceContext.js");
+
+const usePWAUpdateState = () => {
+  const offlineInterface = (0, _OfflineInterfaceContext.useOfflineInterface)();
+  const [updateAvailable, setUpdateAvailable] = (0, _react.useState)(false);
+  const [clientsCount, setClientsCount] = (0, _react.useState)(null);
+
+  const onConfirmUpdate = () => {
+    offlineInterface.useNewSW();
+  };
+
+  const onCancelUpdate = () => {
+    setClientsCount(null);
+  };
+
+  const confirmReload = () => {
+    offlineInterface.getClientsInfo().then(_ref => {
+      let {
+        clientsCount
+      } = _ref;
+
+      if (clientsCount === 1) {
+        // Just one client; go ahead and reload
+        onConfirmUpdate();
+      } else {
+        // Multiple clients; warn about data loss before reloading
+        setClientsCount(clientsCount);
+      }
+    }).catch(reason => {
+      // Didn't get clients info
+      console.warn(reason); // Go ahead with confirmation modal with `0` as clientsCount
+
+      setClientsCount(0);
+    });
+  };
+
+  (0, _react.useEffect)(() => {
+    offlineInterface.checkForNewSW({
+      onNewSW: () => {
+        setUpdateAvailable(true);
+      }
+    });
+  }, [offlineInterface]);
+  const confirmationRequired = clientsCount !== null;
+  return {
+    updateAvailable,
+    confirmReload,
+    confirmationRequired,
+    clientsCount,
+    onConfirmUpdate,
+    onCancelUpdate
+  };
+};
+
+exports.usePWAUpdateState = usePWAUpdateState;
\ No newline at end of file
diff --git a/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js b/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js
index ea02df0..5f67510 100644
--- a/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js
+++ b/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js
@@ -130,10 +130,12 @@ function setUpServiceWorker() {
 
       return !entryShouldBeExcluded;
     });
-    (0, _workboxPrecaching.precacheAndRoute)(restOfManifest); // Same thing for built plugin assets
+    (0, _workboxPrecaching.precacheAndRoute)(restOfManifest); // Same thing for built plugin and pluginified app assets
 
     const pluginPrecacheManifest = self.__WB_PLUGIN_MANIFEST || [];
-    (0, _workboxPrecaching.precacheAndRoute)(pluginPrecacheManifest); // Similar to above; manifest injection from `workbox-build`
+    (0, _workboxPrecaching.precacheAndRoute)(pluginPrecacheManifest);
+    const pluginifiedAppPrecacheManifest = self.__WB_PLUGINIFIED_APP_MANIFEST || [];
+    (0, _workboxPrecaching.precacheAndRoute)(pluginifiedAppPrecacheManifest); // Similar to above; manifest injection from `workbox-build`
     // Precaches all assets in the shell's build folder except in `static`
     // (which CRA's workbox-webpack-plugin handle smartly).
     // Additional files to precache can be added using the
diff --git a/node_modules/@dhis2/pwa/build/es/index.js b/node_modules/@dhis2/pwa/build/es/index.js
index 960e2b2..4c8243f 100644
--- a/node_modules/@dhis2/pwa/build/es/index.js
+++ b/node_modules/@dhis2/pwa/build/es/index.js
@@ -2,3 +2,5 @@ export { setUpServiceWorker } from './service-worker/set-up-service-worker.js';
 export { OfflineInterface } from './offline-interface/offline-interface.js';
 export { checkForUpdates, checkForSWUpdateAndReload, getRegistrationState, REGISTRATION_STATE_UNREGISTERED, REGISTRATION_STATE_WAITING, REGISTRATION_STATE_ACTIVE, REGISTRATION_STATE_FIRST_ACTIVATION } from './lib/registration.js';
 export { getBaseUrlByAppName, setBaseUrlByAppName } from './lib/base-url-db.js';
+export { OfflineInterfaceProvider, useOfflineInterface } from './react/OfflineInterfaceContext.js';
+export { usePWAUpdateState } from './react/usePWAUpdateState.js';
\ No newline at end of file
diff --git a/node_modules/@dhis2/pwa/build/es/react/OfflineInterfaceContext.js b/node_modules/@dhis2/pwa/build/es/react/OfflineInterfaceContext.js
new file mode 100644
index 0000000..9e4705a
--- /dev/null
+++ b/node_modules/@dhis2/pwa/build/es/react/OfflineInterfaceContext.js
@@ -0,0 +1,27 @@
+import PropTypes from 'prop-types';
+import React, { createContext, useContext } from 'react';
+import { OfflineInterface } from '../offline-interface/offline-interface.js';
+const OfflineInterfaceContext = /*#__PURE__*/createContext(null); // nb: instantiating an offline interface here creates a side-effect where
+// it is bundled in the service worker, which causes an error when the `window`
+// API is referenced.
+// It seems like some react and prop types leaks into the SW as well, which is
+// interesting because other exports like from registration.js don't
+
+let theOfflineInterface = null;
+export const OfflineInterfaceProvider = _ref => {
+  let {
+    children
+  } = _ref;
+
+  if (theOfflineInterface === null) {
+    theOfflineInterface = new OfflineInterface();
+  }
+
+  return /*#__PURE__*/React.createElement(OfflineInterfaceContext.Provider, {
+    value: theOfflineInterface
+  }, children);
+};
+OfflineInterfaceProvider.propTypes = {
+  children: PropTypes.node
+};
+export const useOfflineInterface = () => useContext(OfflineInterfaceContext);
\ No newline at end of file
diff --git a/node_modules/@dhis2/pwa/build/es/react/usePWAUpdateState.js b/node_modules/@dhis2/pwa/build/es/react/usePWAUpdateState.js
new file mode 100644
index 0000000..d4fa0ba
--- /dev/null
+++ b/node_modules/@dhis2/pwa/build/es/react/usePWAUpdateState.js
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+import { useOfflineInterface } from './OfflineInterfaceContext.js';
+export const usePWAUpdateState = () => {
+  const offlineInterface = useOfflineInterface();
+  const [updateAvailable, setUpdateAvailable] = useState(false);
+  const [clientsCount, setClientsCount] = useState(null);
+
+  const onConfirmUpdate = () => {
+    offlineInterface.useNewSW();
+  };
+
+  const onCancelUpdate = () => {
+    setClientsCount(null);
+  };
+
+  const confirmReload = () => {
+    offlineInterface.getClientsInfo().then(_ref => {
+      let {
+        clientsCount
+      } = _ref;
+
+      if (clientsCount === 1) {
+        // Just one client; go ahead and reload
+        onConfirmUpdate();
+      } else {
+        // Multiple clients; warn about data loss before reloading
+        setClientsCount(clientsCount);
+      }
+    }).catch(reason => {
+      // Didn't get clients info
+      console.warn(reason); // Go ahead with confirmation modal with `0` as clientsCount
+
+      setClientsCount(0);
+    });
+  };
+
+  useEffect(() => {
+    offlineInterface.checkForNewSW({
+      onNewSW: () => {
+        setUpdateAvailable(true);
+      }
+    });
+  }, [offlineInterface]);
+  const confirmationRequired = clientsCount !== null;
+  return {
+    updateAvailable,
+    confirmReload,
+    confirmationRequired,
+    clientsCount,
+    onConfirmUpdate,
+    onCancelUpdate
+  };
+};
\ No newline at end of file
diff --git a/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js b/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js
index 94d1d55..4d1ad31 100644
--- a/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js
+++ b/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js
@@ -115,10 +115,12 @@ export function setUpServiceWorker() {
       const entryShouldBeExcluded = CRA_MANIFEST_EXCLUDE_PATTERNS.some(pattern => pattern.test(e.url));
       return !entryShouldBeExcluded;
     });
-    precacheAndRoute(restOfManifest); // Same thing for built plugin assets
+    precacheAndRoute(restOfManifest); // Same thing for built plugin and pluginified app assets
 
     const pluginPrecacheManifest = self.__WB_PLUGIN_MANIFEST || [];
-    precacheAndRoute(pluginPrecacheManifest); // Similar to above; manifest injection from `workbox-build`
+    precacheAndRoute(pluginPrecacheManifest);
+    const pluginifiedAppPrecacheManifest = self.__WB_PLUGINIFIED_APP_MANIFEST || [];
+    precacheAndRoute(pluginifiedAppPrecacheManifest); // Similar to above; manifest injection from `workbox-build`
     // Precaches all assets in the shell's build folder except in `static`
     // (which CRA's workbox-webpack-plugin handle smartly).
     // Additional files to precache can be added using the
